src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/Frame.cs

```diff
@@ -374,6 +377,11 @@ public void Reset()
 
             _responseBytesWritten = 0;
             _requestCount++;
+
+            // TODO: probably not the right place to complete
+            RequestBodyPipe.Reader.Complete();
+            RequestBodyPipe.Writer.Complete();
+            RequestBodyPipe.Reset();
```

Comment:
The call the Reset() should go here. Completing the reader/writer should be localized to where the reading/writing occurs as much as possible.